enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,225 @@
+- the cluster capability is a common interface for making OpenShift cluster components optional
+- the cluster capability is already implemented for some operators used by HyperShift (e.g. the console operator)
+- the capability provides a clear API which can be used by any party to understand whether the ingress is enabled
+- most importantly: [the ingress operator's APIs](#ingress-operator-apis) are still created by the cluster version operator
+
+#### Standalone OpenShift
+
+Making the ingress operator optional for the standalone OpenShift is a bigger scope than for HyperShift.
+The changes impact the components which are not used (or used differently) in HyperShift like the installer, the authentication and monitoring operators. To prevent the scope creep for [RFE-3395](https://issues.redhat.com/browse/RFE-3395), the installer should forbid the installation without the ingress capability in [the baseline or additional capabilities of the install config](https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L479-L491).
+No change to the capability API is needed, a simple `if` can ensure this pre-requisite.
+
+#### Ingress operator APIs
+
+The following APIs are not available if the ingress capability is disabled:
+- CRDs which are part of the cluster operator's manifest payload:
+  - [IngressController](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition.yaml)
+  - [DNSRecord](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition-internal.yaml)
+- CRDs which are created at runtime if `gateway-api` feature gate is set:
+  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
+  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
+  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
+
+#### Related APIs
+
+The following APIs are left intact if the ingress capability is disabled:
+- [Route API](https://docs.openshift.com/container-platform/4.13/rest_api/network_apis/route-route-openshift-io-v1.html). Implemented in two places: OpenShift API as [CRD](https://github.com/openshift/api/blob/master/route/v1/route.crd.yaml) and in [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+- [Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html). Implemented in OpenShift API as [CRD](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_ingress.crd.yaml).
+
+However the implementation of these APIs (or parts of them) depend on the ingress operator:
```

Comment:
so why are they going to be left intact?  what is the technical obstacle to disabling them?  (this is covered somewhat under alternatives, i'm not sure there's enough there to convince a reader that serving up apis that don't do anything is the better choice)

Reply:
> what is the technical obstacle to disabling them?

No, not really. Yes, I tried to explain it in the alternatives section. I just updated "Add Route API to the ingress capability" section trying to do a better job at convincing the reader.